### PR TITLE
Improved references to typeof operator in various languages

### PR DESCRIPTION
**Pull request by svick**
* Removed references to C++ `typeid`, since C++/CLI is a niche language.
* Changed incorrect links to `GetType` for VB by making it clear it refers to the `GetType` operator.

    It was very confusing that the documentation of `Type.GetType` said that you should use `GetType` (with a link to `Type.GetType`) instead of `Type.GetType`, when it was supposed to be the `GetType` operator.
* In one case, added missing `typeof`.

### CODE REVIEW

File: xml/System/Type.xml [hunk 1]
```diff
@@ -145,7 +145,7 @@
    
   
 ## Examples  
- The following  example shows a few representative features of <xref:System.Type>. The C# `typeof` operator (`GetType` operator in Visual Basic, `typeid` operator in Visual C++) is used to get a <xref:System.Type> object representing <xref:System.String>. From this <xref:System.Type> object, the <xref:System.Type.GetMethod%2A> method is used to get a <xref:System.Reflection.MethodInfo> representing the <xref:System.String.Substring%2A> overload that takes a starting location and a length.  
+ The following  example shows a few representative features of <xref:System.Type>. The C# `typeof` operator (`GetType` operator in Visual Basic) is used to get a <xref:System.Type> object representing <xref:System.String>. From this <xref:System.Type> object, the <xref:System.Type.GetMethod%2A> method is used to get a <xref:System.Reflection.MethodInfo> representing the <xref:System.String.Substring%2A> overload that takes a starting location and a length.
```
Comment (rpetrusha) [inline xml/System/Type.xml]:
```suggestion
 The following  example shows a few representative features of <xref:System.Type>. The C# `typeof` operator (`GetType` operator in Visual Basic) is used to get a <xref:System.Type> object representing <xref:System.String>. From this <xref:System.Type> object, the <xref:System.Type.GetMethod%2A> method is used to get a <xref:System.Reflection.MethodInfo> representing the <xref:System.String.Substring%2A?displayProperty=nameWithType> overload that takes a starting location and a length.  
```

File: xml/System/Type.xml [hunk 2]
```diff
@@ -7206,7 +7206,7 @@ The current instance or <paramref name="interfaceType" /> argument is an open ge
...
+ You can use the <xref:System.Type.GetType%2A> method to obtain a <xref:System.Type> object for a type in another assembly, if you know its assembly-qualified name which can be obtained from <xref:System.Type.AssemblyQualifiedName>. <xref:System.Type.GetType%2A> causes loading of the assembly specified in `typeName`. You can also load an assembly using the <xref:System.Reflection.Assembly.Load%2A> method, and then use the <xref:System.Type.GetType%2A> or <xref:System.Reflection.Assembly.GetTypes%2A> methods of the <xref:System.Reflection.Assembly> class to get <xref:System.Type> objects. If a type is in an assembly known to your program at compile time, it is more efficient to use `typeof` in C#, or the `GetType` operator in Visual Basic.
```
Comment (rpetrusha) [inline xml/System/Type.xml]:
```suggestion
 You can use the <xref:System.Type.GetType%2A> method to obtain a <xref:System.Type> object for a type in another assembly if you know its assembly-qualified name, which can be obtained from <xref:System.Type.AssemblyQualifiedName>. <xref:System.Type.GetType%2A> causes loading of the assembly specified in `typeName`. You can also load an assembly using the <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> method, and then use the <xref:System.Reflection.Assembly.GetType%2A?displayProperty=nameWithType> or <xref:System.Reflection.Assembly.GetTypes%2A?displayProperty=nameWithType> method to get <xref:System.Type> objects. If a type is in an assembly known to your program at compile time, it is more efficient to use `typeof` in C# or the `GetType` operator in Visual Basic.  
```

File: xml/System/Type.xml [hunk 3]
```diff
@@ -7413,7 +7413,7 @@ Type.GetType("System.Collections.Generic.Dictionary`2[System.String,[MyType,MyAs
...
```
Comment (rpetrusha) [inline xml/System/Type.xml]:
```suggestion
 You can use the <xref:System.Type.GetType%2A> method to obtain a <xref:System.Type> object for a type in another assembly if you know its assembly-qualified name, which can be obtained from <xref:System.Type.AssemblyQualifiedName>. <xref:System.Type.GetType%2A> causes loading of the assembly specified in `typeName`. You can also load an assembly using the <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> method, and then use the <xref:System.Reflection.Assembly.GetType%2A?displayProperty=nameWithType> or <xref:System.Reflection.Assembly.GetTypes%2A?displayProperty=nameWithType> method to get <xref:System.Type> objects. If a type is in an assembly known to your program at compile time, it is more efficient to use `typeof` in C# or the `GetType` operator in Visual Basic.
```

File: xml/System/Type.xml [hunk 4]
```diff
@@ -7642,7 +7642,7 @@ Type.GetType("System.Collections.Generic.Dictionary`2[System.String,[MyType,MyAs
...
```
Comment (rpetrusha) [inline xml/System/Type.xml]:
```suggestion
 You can use the <xref:System.Type.GetType%2A> method to obtain a <xref:System.Type> object for a type in another assembly if you know its assembly-qualified name, which can be obtained from <xref:System.Type.AssemblyQualifiedName>. <xref:System.Type.GetType%2A> causes loading of the assembly specified in `typeName`. You can also load an assembly using the <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> method, and then use the <xref:System.Reflection.Assembly.GetType%2A?displayProperty=nameWithType> or <xref:System.Reflection.Assembly.GetTypes%2A?displayProperty=nameWithType> method to get <xref:System.Type> objects. If a type is in an assembly known to your program at compile time, it is more efficient to use `typeof` in C# or the `GetType` operator in Visual Basic.
```

**Comment by svick** [comment]
@rpetrusha I wasn't able to apply your suggestions on GitHub (it kept saying I should refresh, but refreshing didn't help), so I copy-pasted them manually.

**Comment by rpetrusha** [comment]
Thanks for making the additional changes, @svick, especially since the process didn't go as smoothly as it should have. I'll merge your PR now.

**Pull request merged**